### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by Animan2010** [libgdx/gdx-pay#264]
#### Issue details, reproduction steps/code
On the Android, when the application is restarted multiple times using back button (destroys activity but not application), despite calls to destroy, activity, reportfragment and some google class leaks remain in the memory dump.
If you remove the final attribute from the activity inside class PurchaseManagerGoogleBilling and assign activity to null in the destroy(), then the leaks disappear.

#### Version of gdx-pay and/or relevant dependencies
1.3.7

#### Stacktrace
<img width="1035" alt="image" src="https://github.com/libgdx/gdx-pay/assets/4834983/3e74d168-68d8-4f06-8fd9-db107305a7a6">
<img width="769" alt="image" src="https://github.com/libgdx/gdx-pay/assets/4834983/d4901a35-c177-4b29-b830-8117825534be">

#### Please select the affected platforms and payment service implementation
- [ ] Amazon
- [x] googlebilling
- [ ] googlepay
- [ ] apple robovm
- [ ] Huawei

**Comment by MrStahlfelge** [comment]
Good catch, feel free to open a pr

**Comment by Animan2010** [comment]
> Good catch, feel free to open a pr

I'm not sure that this is a good solution to the problem, rather temporary workaround in my project, it turns out that the PurchaseManagerGoogleBilling instance itself remains registered in memory, it might be better to delete it on destroy.

**Comment by keesvandieren** [comment]
Do you also have the problem in libgdx app as created by gdx setup with
just gdx pay added?

If yes, can you share the project?

Op do 27 jun 2024 om 19:09 schreef Animan2010 ***@***.***>

**Comment by Animan2010** [comment]
> If yes, can you share the project?

Of course, I created an almost empty project using the current gdx-setup and gdx-pay, the result is the same, leaks if restarting the game on the Android emulator (api 28) multiple using the back key.

https://github.com/Animan2010/GdxTest

<img width="886" alt="image" src="https://github.com/libgdx/gdx-pay/assets/4834983/2a45f545-e0f8-4111-830e-71317c682094">
